### PR TITLE
[docker-dhcp-relay] limit privileged flag for dhcp_relay container

### PR DESCRIPTION
**Pull request by maipbui**
#### Why I did it
HLD implementation: Container Hardening (https://github.com/sonic-net/SONiC/pull/1364)
##### Work item tracking
- Microsoft ADO **(number only)**: 14807420

#### How I did it
Reduce linux capabilities in privileged flag

#### How to verify it
Run dhcprelay sonic-mgmt tests
Check container's settings: Privileged is false and container only has default Linux caps, does not have extended caps.
```
admin@vlab-05:~$ docker inspect dhcp_relay | grep Privilege
            "Privileged": false,
admin@vlab-05:~$ docker exec -it dhcp_relay bash
root@vlab-05:/# capsh --print
Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap=ep
Bounding set =cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: rules/docker-dhcp-relay.mk [hunk 1]
```diff
@@ -45,7 +45,7 @@ endif
 endif
 
 $(DOCKER_DHCP_RELAY)_CONTAINER_NAME = dhcp_relay
-$(DOCKER_DHCP_RELAY)_CONTAINER_PRIVILEGED = true
+$(DOCKER_DHCP_RELAY)_CONTAINER_PRIVILEGED = false
```
Comment (Yarden-Z) [inline rules/docker-dhcp-relay.mk:48]: Quick question - why is this container different than the others?
In the others we have the run_opt as the flag defining the privileged state of the container.

Comment (maipbui) [reply rules/docker-dhcp-relay.mk:48]: I think it's related to this PR, making dhcprelay a sonic application extension, https://github.com/sonic-net/sonic-buildimage/pull/6531/files, HLD https://github.com/sonic-net/SONiC/blob/master/doc/sonic-application-extension/sonic-application-extention-hld.md

**Comment by qiluo-msft** [comment]
In PR description "How to verify it" section, could you also check the container's privilege setting and paste the result?

**Comment by maipbui** [comment]
> In PR description "How to verify it" section, could you also check the container's privilege setting and paste the result?

Updated

**Review by qiluo-msft — APPROVED** [review]

**Review by kellyyeh — APPROVED** [review]

**Pull request merged**